crates/uv-resolver/src/lock/mod.rs

```diff
@@ -2837,6 +2837,35 @@ struct PackageMetadata {
     dependency_groups: BTreeMap<GroupName, BTreeSet<Requirement>>,
 }
 
+impl PackageMetadata {
+    fn unwire(self, requires_python: &RequiresPython) -> Result<PackageMetadata, LockError> {
+        // We need to complexify these markers so things like
+        // `requires_python < '0'` get normalized to False
+        let unwire_requirements =
+            |requirements: BTreeSet<Requirement>| -> Result<BTreeSet<Requirement>, LockError> {
+                requirements
+                    .into_iter()
+                    .map(|mut requirement| {
+                        let complexified_marker =
+                            requires_python.complexify_markers(requirement.marker);
+                        requirement.marker = complexified_marker;
+                        Ok(requirement)
+                    })
+                    .collect()
+            };
+
+        Ok(PackageMetadata {
+            requires_dist: unwire_requirements(self.requires_dist)?,
```

Comment:
This is technically a drive-by fix; ideally I write a test that shows this was also broken.